src/Services/Import/CSV/ImportCSV.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services\Import\CSV;
+
+use App\Services\Import\Import;
+use App\Services\Import\ImportRequest;
+use App\Services\Import\Savers\Saver;
+use League\Csv\Reader;
+use League\Csv\SyntaxError;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+class ImportCSV extends Import
+{
+    private static array $headerTitles = ['Product Code', 'Product Name', 'Product Description', 'Stock', 'Cost in GBP', 'Discontinued'];
+
+    private bool $headerMustSynchronize;
+    private Reader $csv;
+
+    public function __construct(
+        array $files,
+        CSVSettings $csvSettings,
+        bool $isTest,
+        Saver $saver = null,
+    ) {
+        $data = $this->setDataFromFiles($files, $csvSettings);
```

Comment:
in this case, you will use the same CSV settings for different files. What in files you will have to use different delimiters?

Reply:
added the ability to configure each import